gradle/render-javadoc.gradle

```diff
@@ -39,394 +35,398 @@ allprojects {
+      linkedProjectNames.add(otherProjectPath)
+    }
+  }
+
+  @TaskAction
+  public void render() {
+    def javadocCmd = org.gradle.internal.jvm.Jvm.current().getJavadocExecutable()
+
+    def srcDirs = srcDirSet.srcDirs.findAll { dir -> dir.exists() }
+    def optionsFile = project.file("${getTemporaryDir()}/javadoc-options.txt")
+    
+    def opts = []
+    opts << [ '-overview', project.file("${srcDirs[0]}/overview.html") ]
+    opts << [ '-sourcepath', srcDirs.join(File.pathSeparator) ]
+    opts << [ '-subpackages', project.path.startsWith(':lucene') ? 'org.apache.lucene' : 'org.apache.solr' ]
+    opts << [ '-d', outputDir ]
+    opts << '-protected'
+    opts << [ '-encoding', 'UTF-8' ]
+    opts << [ '-charset', 'UTF-8' ]
+    opts << [ '-docencoding', 'UTF-8' ]
+    opts << '-noindex'
+    opts << '-author'
+    opts << '-version'
+    if (linksource) {
+      opts << '-linksource'
+    }
+    opts << '-use'
+    opts << [ '-locale', 'en_US' ]
+    opts << [ '-windowtitle', title ]
+    opts << [ '-doctitle', title ]
+    if (!classpath.isEmpty()) {
+      opts << [ '-classpath', classpath.asPath ]
+    }
+    opts << [ '-bottom', "<i>Copyright &copy; 2000-${project.buildYear} Apache Software Foundation. All Rights Reserved.</i>" ]
+
+    opts << [ '-tag', 'lucene.experimental:a:WARNING: This API is experimental and might change in incompatible ways in the next release.' ]
+    opts << [ '-tag', 'lucene.internal:a:NOTE: This API is for internal purposes only and might change in incompatible ways in the next release.' ]
+    opts << [ '-tag', "lucene.spi:t:SPI Name (case-insensitive: if the name is 'htmlStrip', 'htmlstrip' can be used when looking up the service)." ]
+
+    def allOfflineLinks = [:]
+    allOfflineLinks.putAll(offlineLinks)
+
+    // Resolve inter-project links.
+    linkedProjects.collect { otherProject ->
+      // For relative links we compute the actual relative link between projects.
+      // TODO: if we don't want relative Solr-Lucene links we need to add a condition here
+      // and it'll just render as an offline link.
+      if (relativeProjectLinks) {
+        def pathTo = otherProject.tasks[name].outputDir.toPath().toAbsolutePath()
+        def pathFrom = outputDir.toPath().toAbsolutePath()
+        def relative = pathFrom.relativize(pathTo).toString().replace(File.separator, '/')
+        opts << [ '-link', relative ]
+      } else {
+        // For absolute links, we determine the target URL by assembling the full URL.
+        def base = otherProject.path.startsWith(":lucene") ? luceneDocUrl : solrDocUrl
+        allOfflineLinks.put("${base}/${otherProject.relativeDocPath}/".toString(), otherProject.tasks[name].outputDir)
+      }
+    }
+
+    // Add offline links.
+    // TODO: add some sanity check/ validation here to ensure folder/package-list or folder/element-list is present?
+    allOfflineLinks.each { url, folder ->
+      opts << [ '-linkoffline', url, folder ]
+    }
```

Comment:
Would it be better to use Unix terminology "dir" or "directory" for variable names ?

Reply:
Sure, I don't mind. :) Sorry for lack of consistency.